File: MyCity4Kids/MyCity4Kids/src/com/mycity4kids/models/CollectionsModels/UpdateCollectionRequestModel.java
package com.mycity4kids.models.CollectionsModels;

import java.util.ArrayList;

public class UpdateCollectionRequestModel {
    private String name;
    private String userId;
    private ArrayList<String> userCollectionId;
    private String item;
    private String imageUrl;
    private String itemType;
    private Boolean deleted = false;
    private Boolean isPublic = false;

    public Boolean getPublic() {
        return isPublic;
    }

    public void setPublic(Boolean aPublic) {
        isPublic = aPublic;
    }

    public Boolean getDeleted() {
        return deleted;
    }

    public void setDeleted(Boolean deleted) {
        this.deleted = deleted;
    }

    public String getImageUrl() {
        return imageUrl;
    }

    public void setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
    }

    public String getItemType() {
        return itemType;
    }

    public void setItemType(String itemType) {
        this.itemType = itemType;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public ArrayList<String> getUserCollectionId() {
        return userCollectionId;
    }

    public void setUserCollectionId(ArrayList<String> userCollectionId) {
        this.userCollectionId = userCollectionId;
    }

    public String getItem() {
        return item;
    }

    public void setItem(String item) {
        this.item = item;
    }
}

File: app/src/main/java/com/mycity4kids/models/collectionsModels/UpdateCollectionRequestModel.kt
package com.mycity4kids.models.collectionsModels

import com.google.gson.annotations.SerializedName


class UpdateCollectionRequestModel {
    @SerializedName("name")
    var name: String? = null
    @SerializedName("userId")
    var userId: String? = null
    @SerializedName("userCollectionId")
    var userCollectionId: ArrayList<String>? = null
    @SerializedName("item")
    var item: String? = null
    @SerializedName("imageUrl")
    var imageUrl: String? = null
    @SerializedName("itemType")
    var itemType: String? = null
    @SerializedName("deleted")
    var deleted: Boolean? = false
    @SerializedName("isPublic")
    var isPublic: Boolean? = null
    @SerializedName("summary")
    var summary: String? = null

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as UpdateCollectionRequestModel

        if (item != other.item) return false

        return true
    }

    override fun hashCode(): Int {
        return item?.hashCode() ?: 0
    }


}
